common/src/main/java/org/opensearch/ml/common/output/model/ModelTensors.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.ml.common.output.model;
+
+import lombok.Builder;
+import lombok.Getter;
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.common.io.stream.Writeable;
+import org.opensearch.common.xcontent.ToXContentObject;
+import org.opensearch.common.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+@Getter
+public class ModelTensors implements Writeable, ToXContentObject {
+    public static final String OUTPUT_FIELD = "output";
+    private List<ModelTensor> mlModelTensors;
+
+    @Builder
+    public ModelTensors(List<ModelTensor> mlModelTensors) {
+        this.mlModelTensors = mlModelTensors;
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject();
+        if (mlModelTensors != null && mlModelTensors.size() > 0) {
+            builder.startArray(OUTPUT_FIELD);
+            for (ModelTensor output : mlModelTensors) {
+                output.toXContent(builder, params);
+            }
+            builder.endArray();
+        }
+        builder.endObject();
+        return builder;
+    }
+
+    public ModelTensors(StreamInput in) throws IOException {
+        if (in.readBoolean()) {
+            mlModelTensors = new ArrayList<>();
+            int size = in.readInt();
+            for (int i=0; i<size; i++) {
+                mlModelTensors.add(new ModelTensor(in));
+            }
+        }
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        if (mlModelTensors != null && mlModelTensors.size() > 0) {
+            out.writeBoolean(true);
+            out.writeInt(mlModelTensors.size());
+            for (ModelTensor output : mlModelTensors) {
+                output.writeTo(out);
+            }
+        } else {
+            out.writeBoolean(false);
+        }
+    }
+
+    public void filter(ModelResultFilter resultFilter) {
+        boolean returnBytes = resultFilter.isReturnBytes();
+        boolean returnNUmber = resultFilter.isReturnNumber();
```

Comment:
returnNUmber -> returnNumber?

Reply:
As this PR has been approved. I will address these typos in next PR.